test/test_cuda.py

```diff
@@ -1108,6 +1108,49 @@ def test_inplace_unary_mem_overlap(self):
     def test_inplace_binary_mem_overlap(self):
         _TestTorchMixin._test_inplace_binary_mem_overlap(self, device='cuda')
 
+    # This test does not work until input/output overlap check is implemented on CUDA.
+    # Remove @unittest.expectedFailure when it's done.
+    @unittest.expectedFailure
+    def test_unary_op_input_output_overlap(self):
+        sz = 3
+        doubles = torch.randn(2 * sz, device='cuda')
+        positives = torch.randint(1, 100, (2 * sz,), device='cuda').double()
+        ints = torch.randint(-100, 100, (2 * sz,), device='cuda')
+
+        _TestTorchMixin._test_unary_op_overlap(self, sz, doubles, torch.abs)
```

Comment:
nit: You can store an array of named tuples (size, dtype, op) and then do a for-loop to avoid writing `_TestTorchMixin._test_unary_op_overlap` again and again

Reply:
addressed